app/qml/NavigationPanel.qml

```diff
@@ -11,222 +11,390 @@ import QtQuick 2.14
 import QtQuick.Layouts 1.14
 import QtQuick.Controls 2.14
 import QtGraphicalEffects 1.0
+import QtQuick.Shapes 1.14
 
 import "."  // import InputStyle singleton
+import "./components" as Components
+import "./map" as Map
 
-Item {
-    id: navigationPanel
-    property real rowHeight: InputStyle.rowHeight
+import lc 1.0
 
-    property var navigationTargetFeature
+Item {
+  id: root
+
+  readonly property alias isOpen: drawer.opened
+
+  property real featureToGpsDistance: -1
+  property var navigationTargetFeature
+
+  property var mapCanvas
+
+  property bool autoFollow: true
+  property var calculatedNavigationExtent
+  property string mapStateBeforeNavigation
+
+  property real distanceThresholdToShortMode: 1 // in metres
```

Comment:
just wondering if this should be hardcoded

Reply:
Let's keep it hard-coded for the time being... it will be easy to add configuration to the settings later if needed :-)